server/src/main/java/org/jboss/as/server/moduleservice/ModuleLoadService.java

```diff
@@ -129,6 +129,16 @@ public Module getValue() throws IllegalStateException, IllegalArgumentException
     }
 
     private static ServiceName install(final ServiceTarget target, final ModuleIdentifier identifier, ModuleLoadService service) {
+        for (ModuleDependency dependency : service.allDependencies) {
+            if (dependency.getReason().isPresent() && !dependency.isUserSpecified() && dependency.getIdentifier() != null
+                    && dependency.getIdentifier().getName() != null && dependency.getIdentifier().getSlot() != null) {
+                ServerLogger.AS_ROOT_LOGGER.logImplicitDependency(dependency.getIdentifier().getName(), dependency.getIdentifier().getSlot(), dependency.getReason().get());
+            } else if (!dependency.isUserSpecified() && dependency.getIdentifier() != null && dependency.getIdentifier().getName() != null
+                    && dependency.getIdentifier().getSlot() != null) {
+                ServerLogger.AS_ROOT_LOGGER.logImplicitDependencyWithoutReason(dependency.getIdentifier().getName(), dependency.getIdentifier().getSlot());
+            }
```

Comment:
Let's try to avoid this, this is a bit convoluted code that could be even inside of the ModuleDependency code.

I've found we have already logged the dependencies added into the deployments. Check this log:
https://github.com/wildfly/wildfly-core/blob/main/server/src/main/java/org/jboss/as/server/deployment/module/ModuleSpecProcessor.java#L356

We can use it to log also the reasons by using the ModuleDependency.toString().